library/src/commonMain/kotlin/com/badoo/reaktive/disposable/DisposableWrapper.kt

```diff
@@ -0,0 +1,50 @@
+package com.badoo.reaktive.disposable
+
+import com.badoo.reaktive.utils.lock.newLock
+import com.badoo.reaktive.utils.lock.synchronized
+
+/**
+ * Thread-safe container of one [Disposable]
+ */
+class DisposableWrapper(
+    private var disposable: Disposable? = null
+) : Disposable {
+
+    private val lock = newLock()
+    private var _isDisposed = false
```

Comment:
Shouldn't this be @ Volatile?

Reply:
Don't think so, it's only accessed from synchronized block